include/app/action.hpp

```diff
@@ -81,4 +90,29 @@ class Action : public QObject {
 
     QString name_;
     QString key_;
+    
+};
+
+class ActionEventFilter : public QObject
+{
+    Q_OBJECT
+    public:
+        ActionEventFilter(){};
+        bool eventFilter(QObject* obj, QEvent* event);
+        static ActionEventFilter *get_instance();
```

Comment:
same thing as above lets try to get away from singletons

since this seems AA specific(?) it probs be good to put it in Session::AndroidAuto (or wherever you decide to add the aa handler to)

Reply:
Not AA specific - all single key shortcuts go through event filter now

Reply:
okay then it can probs go in Session::System (same place where all the action stuff is currently happening)